Subject: [PATCH] refpolicy: MLS - sysadm rw to clearance

The two new rules make sysadm_t domain MLS trusted for:
 - reading from files up to its clearance(s0-s15).
 - writing to processes up to its clearance(s0-s15).

With default MLS policy, root user would login in as sysadm_t:s0 by
default. Most processes will run in sysadm_t:s0 because no
domtrans/rangetrans rules, as a result, even root could not access
high level files/processes.

So with the two new rules, root user could work easier in MLS policy.

Upstream-Status: Inappropriate [only for Poky]

Signed-off-by: Xin Ouyang <Xin.Ouyang@windriver.com>
---
 policy/modules/roles/sysadm.te | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/policy/modules/roles/sysadm.te b/policy/modules/roles/sysadm.te
index cee455d..85ff145 100644
--- a/policy/modules/roles/sysadm.te
+++ b/policy/modules/roles/sysadm.te
@@ -29,6 +29,9 @@ corecmd_exec_shell(sysadm_t)
 
 mls_process_read_up(sysadm_t)
 
+mls_file_read_to_clearance(sysadm_t)
+mls_process_write_to_clearance(sysadm_t)
+
 ubac_process_exempt(sysadm_t)
 ubac_file_exempt(sysadm_t)
 ubac_fd_exempt(sysadm_t)
-- 
1.7.11.7

